tests/queries/0_stateless/02710_allow_suspicious_indices.sql

```diff
@@ -0,0 +1,20 @@
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32) ENGINE = MergeTree() order by (id, id) settings allow_suspicious_indices = 1;
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32) ENGINE = MergeTree() order by (id, id) settings allow_suspicious_indices = 0;  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax GRANULARITY 1) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 1;
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 1;
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax GRANULARITY 1) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 0;  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 0;  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
```

Comment:
About `AlTER TABLE`: Is `ADD INDEX` also checked? --> https://clickhouse.com/docs/en/sql-reference/statements/alter/skipping-index

Reply:
Added